hcloud/client_handler_retry.go

```diff
@@ -30,13 +34,53 @@ func (h *retryHandler) Do(req *http.Request, v any) (resp *Response, err error)
 			// - request preparation
 			// - network connectivity
 			// - http status code (see [errorHandler])
-			if IsError(err, ErrorCodeConflict) {
-				time.Sleep(h.backoffFunc(retries))
-				retries++
-				continue
+			if ctx.Err() != nil {
+				return resp, ctx.Err()
+			}
+
+			if retries < h.maxRetries && retryPolicy(resp, err) {
+				select {
+				case <-ctx.Done():
+					return resp, err
+				case <-time.After(h.backoffFunc(retries)):
+					retries++
+					continue
+				}
 			}
 		}
 
 		return resp, err
 	}
 }
+
+func retryPolicy(resp *Response, err error) bool {
+	if err != nil {
+		var apiErr Error
+		var netErr net.Error
+
+		switch {
+		case errors.As(err, &apiErr):
+			switch apiErr.Code { //nolint:exhaustive
+			case ErrorCodeConflict, ErrorCodeLocked:
```

Comment:
I am actually not sure if we should retry on "locked" errors. Many errors are used in many unexpected situations, we therefor cannot really rely on most errors codes.

Because we cannot rely on the error code, we must use a retry helper to be able to retry in a specific context.

If we want to add this here, we first must change the API to be consistent in how the error codes are used.